src/NServiceBus.Core.Analyzer/FeatureDefaultsEnableFeatureAnalyzer.cs

```diff
@@ -0,0 +1,122 @@
+namespace NServiceBus.Core.Analyzer
+{
+    using System.Collections.Immutable;
+    using System.Linq;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using Microsoft.CodeAnalysis.Diagnostics;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class FeatureDefaultsEnableFeatureAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(diagnostic);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(startContext =>
+            {
+                var featureType = startContext.Compilation.GetTypeByMetadataName("NServiceBus.Features.Feature");
+                var settingsExtensionsType = startContext.Compilation.GetTypeByMetadataName("NServiceBus.Features.SettingsExtensions");
+
+                if (featureType == null || settingsExtensionsType == null)
+                {
+                    return;
+                }
+
+                startContext.RegisterSyntaxNodeAction(
+                    ctx => AnalyzeEnableInvocation(ctx, featureType, settingsExtensionsType),
+                    SyntaxKind.InvocationExpression);
+            });
+        }
+
+        static void AnalyzeEnableInvocation(
+            SyntaxNodeAnalysisContext context,
+            INamedTypeSymbol featureType,
+            INamedTypeSymbol settingsExtensionsType)
+        {
+            if (context.Node is not InvocationExpressionSyntax invocation)
+            {
+                return;
+            }
+
+            if (context.SemanticModel.GetSymbolInfo(invocation, context.CancellationToken).Symbol is not IMethodSymbol methodSymbol)
```

Comment:
Before an expensive query on the semantic model, we need to check the properties of the syntax tree to see if the Identifier.Name (or something like that) is "EnableFeature". Otherwise we'll look up the methodSymbol for every single invocation in the compilation.